cruise.umple/test/cruise/umple/statemachine/implementation/java/stateDependentMethodDeclaration_2.java.txt

```diff
@@ -0,0 +1,237 @@
+/*PLEASE DO NOT EDIT THIS CODE*/
+/*This code was generated using the UMPLE ${last.version} modeling language!*/
+
+package example;
+
+// line 3 "../stateDependentMethodDeclaration_2.ump"
+public class X
+{
+
+  //------------------------
+  // MEMBER VARIABLES
+  //------------------------
+
+  //X State Machines
+  public enum Sm { init, going }
+  private Sm sm;
+
+  //------------------------
+  // CONSTRUCTOR
+  //------------------------
+
+  public X()
+  {
+    setSm(Sm.init);
+  }
+
+  //------------------------
+  // INTERFACE
+  //------------------------
+
+  public String getSmFullName()
+  {
+    String answer = sm.toString();
+    return answer;
+  }
+
+  public Sm getSm()
+  {
+    return sm;
+  }
+
+  public void pstatus()
+  {
+    switch(sm)
+    {
+      case init:
+        System.out.println("init");

```

Comment:
Would probably be a good idea to test injections for methods with return values as well

Reply:
Yes,I agee